NewBatteryState.h

```diff
@@ -0,0 +1,39 @@
+#include "Battery.h"
+class NewBatteryState : public Battery
+{
+  public:
+    bool checkTempRange(float temperature)
+    {
+        if (temperature < 0 || temperature > 45)
+        {
+            cout << "Temperature out of range!\n";
+            return false;
+        }
+        else
+            return true;
+    }
+    bool checkSocRange(float soc)
+    {
+        if (soc < 20 || soc > 80)
+        {
+            cout << "State of Charge out of range!\n";
+            return false;
+        }
+        else
+            return true;
+    }
+    bool checkChargeRateRange(float chargeRate)
+    {
+        if (chargeRate > 0.8)
+        {
+            cout << "Charge Rate out of range!\n";
+            return false;
+        }
+        else
+            return true;
+    }
+    bool checkBatteryIsOk(float t, float s, float c)
```

Comment:
naming of the parameters can be improved , example renaming 't' to 'temperature' would provide clarity on what is expected.

Reply:
naming updated